Make Jupyter Notebook 4.x the minimum baseline

parente opened the pull request:
Fixes issue #49

Merging pending ability to test and update declarative widget dashboard examples based on jupyter-incubator/declarativewidgets#67

parente commented:
@lbustelo Have a look at this PR for all the Makefile, etc. updates. Widgets are going to differ of course, but it's a starting point. I'll hold this PR here until widgets have a 4.x compatibility PR that I can finish testing all our widget dashboard examples against.

parente commented:
Rebased today since master has change quite a bit since I opened the PR.

wellecks commented on Makefile:
```diff
@@ -1,74 +1,68 @@
-.PHONY: build clean dev help sdist test install remote
+.PHONY: clean dev dev-with-widgets help install js sdist test 
+
+REPO:=jupyter/pyspark-notebook:a388c4a66fd4
```
Why the use of this tag vs jupyter/pyspark-notebook:4.0 ?

parente replied:
https://github.com/jupyter/docker-stacks#stacks-tags-versioning-and-progress

tl;dr: 4.0 is legacy and no longer moving.

parente commented:
Rebased on master (0.2.0.dev). Added two additional commits to account for changes. Manually tested all demos that do not require declarative widgets.

parente commented:
Tested a whole slew of demos using what's in the jupyter-incubator/declarativewidgets#114 PR. Fixed on e error in the dashboard.js (init function now required). Everything else was solid.
